### PR TITLE
Add ack for worker connection

### PR DESCRIPTION
**Pull request by Nosibb**
adding test for worker connection.
Details is issue: https://github.com/locustio/locust/issues/2044

### CODE REVIEW

**Comment by cyberw** [comment]
@heyman What do you think?

Would it be possible to make a test case that really tests the functionality? (maybe by patching master to discard the first connect message)

Is this backwards compatible, assuming an old worker discards the ack? I'm thinking mainly of other worker implementations like @myzhan 's Boomer

**Comment by Nosibb** [comment]
it requires fixes for sure. I'm working on it

File: locust/test/test_runners.py [hunk 1]
```diff
@@ -2321,7 +2322,7 @@ def my_task(self):
             server.mocked_send(Message("client_ready", __version__, "zeh_fake_client2"))
             self.assertEqual(2, len(master.clients))
             sleep(0.1)  # give time for messages to be sent to clients
-            self.assertEqual(2, len(server.outbox))
+            self.assertEqual(4, len(server.outbox))
```
Comment (Nosibb) [inline locust/test/test_runners.py:2325]: there are many changes like this because of ack message

**Review by bomba5-zoetis — APPROVED** [review]

**Comment by cyberw** [comment]
@Nosibb Did you see my response?

**Comment by cyberw** [comment]
@Nosibb ?

**Comment by Nosibb** [comment]
@cyberw 
I remember about this. I was busy :) I'll finish it next week

**Comment by cyberw** [comment]
No rush, I just wanted to make sure you were't waiting for me or something :)

**Comment by Nosibb** [comment]
Let me know what do you think about the new test.
In my opinion, it is backwards compatible. In older versions, we will see a warning. Is it enough?

File: locust/test/test_runners.py [hunk 2]
```diff
@@ -3077,6 +3094,7 @@ def my_task(self):
 
         with mock.patch("locust.rpc.rpc.Client", mocked_rpc()) as client:
             environment = Environment()
+            client.mocked_send(Message("ack", {}, "dummy_client_id"))
```
[the remark this answers is not in the document]
Comment (Nosibb) [reply locust/test/test_runners.py]: you are right. It is now simplified

**Comment by cyberw** [comment]
@heyman Any comments? Looks reasonable to me but I didnt originally write the connection logic.

I’d prefer the constants be named CONNECT_* instead of CONNECTION_* (to indicat that they relate to connection setup, not once connection has been enabled)

**Comment by cyberw** [comment]
Thx! I’ll make sure to bump the minor version when I do the next release.

**Pull request merged**